src/lib.rs

```diff
@@ -41,6 +41,14 @@ pub use middleware::GrantsMiddleware;
 /// async fn role_macro_secured() -> HttpResponse {
 ///     HttpResponse::Ok().body("some secured info")
 /// }
+///
+/// // Additional security condition to ensure the protection of the endpoint
+/// #[has_roles("USER", secure = "user_id==user.id")]
+/// #[get("/resource/{user_id}")]
+/// async fn role_macro_secured_with_params(web::Path(user_id): web::Path<i32>, user: web::Data<User>) -> HttpResponse {
```

Comment:
Add `use actix_web::web` to the beginning of each code block (documentation is also subject to verification)

In this case, you should just append to existing part: 
 `use actix_web::{web, HttpResponse};`